src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -262,7 +262,14 @@ abs_client::abs_client(
   , _client(conf, &as, conf._probe, conf.max_idle_time)
   , _probe(conf._probe) {}
 
-ss::future<> abs_client::stop() { return _client.stop(); }
+ss::future<> abs_client::stop() {
+    vlog(abs_log.debug, "Stopping ABS client");
+
+    co_await _client.stop();
+    co_await _client.wait_input_shutdown();
```

Comment:
I would add a comment with reference to the issue here, to explain why only with ABS we wait

Reply:
Not opposed to adding one, but isn't the commit message enough? When I'm confused by a piece of code, I always refer to it's commit message.

Reply:
The commit message is very good, i would basically use it as the comment. i'm thinking if in the future somebody has a look at the various remote storage clients, it could be confusing that this operations is done only here, and the commit message could be hidden by a refactor